File: igVortex/Python/src/ignVelocityw2.py
import numpy as np
import src.globals as g


def ignVelocityw2(m, ZC, NC, ZF, GAMAw, iGAMAw, eps):
    """
    Normal velocity contribution on the airfoil by the wake vortex.
    Input:
    * m                 # of vortex points.
    * ZC (0, m-1)       collocation points.
    * NC (0, m-1)       unit normal complex number
    * ZF (0, iGAMAw)    location of the wake vorticies (0:iGAMAw)
    * iGAMAw            # of wake vorticies
    Output:
    * VNW               normal vleocity components at the collocation points due to the wake vorticies
    """

    VNW = np.zeros((m - 1))

    if g.ibios == 0:
        eps = eps * 1000

        for i in range(1, m):
            for j in range(1, iGAMAw):
                r = np.abs(ZC[i - 1] - ZF[j - 1])
                GF = complex(0.0, 0.0)
                if r > eps:
                    GF = 1.0 / (ZC[i - 1] - ZF[j - 1])
                VNW[i - 1] = VNW[i - 1] + GAMAw[j - 1] * \
                    np.imag(NC[i - 1] * GF) / (2.0 * np.pi)

    elif g.ibios == 1:
        for i in range(1, m):
            for j in range(1, iGAMAw + 1):
                r = np.abs(ZC[i - 1] - ZF[j - 1])
                if r < eps:
                    GF = complex(0.0, 0.0)
                else:
                    GF = 1.0 / (ZC[i - 1] - ZF[j - 1])
                    if r < g.delta:
                        GF = GF * (r / g.delta) ** 2
                VNW[i - 1] = VNW[i - 1] + GAMAw[j - 1] * \
                    np.imag(NC[i - 1] * GF) / (2.0 * np.pi)

    return VNW, eps

File: 2D Simulation/MATLAB/ignVelocityw2.m
function [VNW] = ignVelocityw2(m,ZC, NC, ZF, GAMAw, iGAMAw)
%Normal velocity contribution on the airfoil by the wake vortex
%INPUT
% m                         # of vortex points
% ZC    (1,m-1)         collocation points
% NC        (1,m-1)         unit normal complex number
% ZF        (1,iGAMAw)    location of the wake vortices (1:iGAMAw)
% GAMAw     (1,iGAMAw)      wake vortex
% iGAMAw                    # of wake vortices
%OUTPUT
% VNW       (1,m-1)         normal velocity components at the collocation point due to the wake vortices

global eps delta ibios

VNW=zeros(1,m-1);
switch ibios
    case 0
        eps=eps*1000;
        for i=1:m-1
            VNW(i)=0.0;
            for j=1:iGAMAw %skipped for iGAMAw=0 in istep=1          
                r=abs(ZC(i)-ZF(j));
                GF=complex(0.0,0.0);
                if r > eps 
                    GF=1.0/(ZC(i)-ZF(j));
                end
                VNW(i)=VNW(i)+GAMAw(j)*imag(NC(i)*GF)/(2.0*pi);
            end
        end
    case 1
        for i=1:m-1
            VNW(i)=0.0;
            for j=1:iGAMAw %skipped for iGAMAw=0 in istep=1          
                r=abs(ZC(i)-ZF(j));
                if r < eps
                    GF=complex(0.0,0.0);
                else 
                    GF=1.0/(ZC(i)-ZF(j));
                    if r < delta
                        GF=GF*(r/delta)^2;
                    end
                end
                VNW(i)=VNW(i)+GAMAw(j)*imag(NC(i)*GF)/(2.0*pi);
            end
        end

end

end

